tests/to_py.rs

```diff
@@ -232,3 +232,23 @@ fn to_pyarray_object_array() {
         }
     })
 }
+
+#[test]
+fn slice_box_type_confusion() {
```

Comment:
As detailed in #232, this test will crash without the fix proposed here as `Py_DECREF` will be called on 1,2,3 interpreted as pointers into the Python heap. The only way I see to make this a proper assert would be to add special PyClass which makes its drops observable and asserting on those not happening. But I think this would be a mouthful while reliably crashing with SIGSEGV is a pretty good "assertion" in itself.

Do you think adding a comment and link to the issue could resolve this?

Reply:
> Do you think adding a comment and link to the issue could resolve this?

Tried adding a reasonably detailed explanation and link to explain the test.

Reply:
That's perfect, thanks 👍